packages/datatrak-web/src/features/Survey/SurveyLayout.tsx

```diff
@@ -0,0 +1,175 @@
+})`
+  flex: 1;
+  margin-left: 1rem;
+  max-width: 63rem;
+  padding: 0;
+  position: relative;
+  display: flex;
+  flex-direction: column;
+`;
+const FormActions = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 1rem;
+  border-top: 1px solid ${props => props.theme.palette.divider};
+
+  button:last-child {
+    margin-left: auto;
+  }
+`;
+
+const ButtonGroup = styled.div`
+  display: flex;
+  button,
+  a {
+    &:last-child {
+      margin-left: 1rem;
+    }
+  }
+`;
+
+/**
+ * This layout is used for the survey screens as well as the review screen.
+ */
+export const SurveyLayout = () => {
+  const [cancelModalOpen, setCancelModalOpen] = useState(false);
+  const navigate = useNavigate();
+  const params = useParams<SurveyParams>();
+  const {
+    setFormData,
+    formData,
+    isLast,
+    screenNumber,
+    sideMenuOpen,
+    numberOfScreens,
+    isReviewScreen,
+  } = useSurveyForm();
+  const formContext = useForm({ defaultValues: formData });
+  const { handleSubmit } = formContext;
+
+  const handleStep = (path, data) => {
+    setFormData({ ...formData, ...data });
+    navigate(path);
+  };
+
+  const onStepPrevious = handleSubmit(data => {
+    let path = ROUTES.SURVEY_SELECT;
+    const prevScreenNumber = isReviewScreen ? numberOfScreens : screenNumber! - 1;
+    if (prevScreenNumber) {
+      path = generatePath(ROUTES.SURVEY_SCREEN, {
+        ...params,
+        screenNumber: String(prevScreenNumber),
+      });
+    }
+
+    handleStep(path, data);
+  });
+
+  const handleSubmitForm = data => {
+    // Placeholder for actual submission logic
+    console.log(data);
+    navigate(ROUTES.SURVEY_SUCCESS);
+  };
+
+  const navigateNext = data => {
+    const path = isLast
+      ? generatePath(ROUTES.SURVEY_REVIEW, params)
+      : generatePath(ROUTES.SURVEY_SCREEN, {
+          ...params,
+          screenNumber: String(screenNumber! + 1),
+        });
+    handleStep(path, data);
+  };
+
+  const handleSubmitScreen = handleSubmit(data => {
+    if (isReviewScreen) return handleSubmitForm(data);
+    return navigateNext(data);
+  });
+
+  const openCancelModal = () => {
+    setCancelModalOpen(true);
+  };
+
+  return (
+    <ScrollableLayout>
+      <FormProvider {...formContext}>
+        <Wrapper>
+          <SurveySideMenu />
+          <Container $sideMenuClosed={!sideMenuOpen && !isReviewScreen}>
+            <Paper>
+              <Outlet />
+              <FormActions>
+                <Button
+                  onClick={onStepPrevious}
+                  startIcon={<ArrowBackIosIcon />}
+                  variant="text"
+                  color="default"
+                >
+                  Back
+                </Button>
+                <ButtonGroup>
+                  <Button onClick={openCancelModal} variant="outlined">
+                    Cancel
+                  </Button>
+                  <Button type="submit" onClick={handleSubmitScreen}>
```

Comment:
If it's not possible, then it's not a big deal but my preference is to use the form onSubmit handler to submit each screen as it's more semantic and accessible and ensures the enter button works. I think it should be easy enough to just add a form element around the `Outlet` or some thing similar.

Reply:
The main reason I did this was because I didn't want to have headings (eg an `h2` inside the `form` element, which would happen if this was the case.

Reply:
oh true that makes sense - that is a bit of a conundrum but having the headings inside the form is probably the lesser of the 2 evils so thanks for the change.